dev/com.ibm.ws.concurrent.persistent_fat_multiple/fat/src/com/ibm/ws/concurrent/persistent/fat/multiple/DatabaseContainerUtil.java

```diff
@@ -0,0 +1,138 @@
+     * @param serv - LibertyServer server instance being used for this FAT suite.
+     * @param cont - JdbcDatabaseContainer instance being used for database connectivity.
+     * @param type - DBContainerType type information about this container.
+     */
+    public static void setupDataSourceProperties(LibertyServer serv, JdbcDatabaseContainer<?> cont, DatabaseContainerType type) throws Exception {
+       	//Get properties from container
+    	String username = cont.getUsername();
+    	String password = cont.getPassword();
+    	String databaseName = null;
+    	String port = Integer.toString(cont.getFirstMappedPort());
+    	String serverName = cont.getContainerIpAddress();
+    	
+    	try {
+    		databaseName = cont.getDatabaseName();
+    	} catch (UnsupportedOperationException e) {
+    		Log.info(c, "setupDataSourceProperties", e.toString());
+    	}
+    	
+    	//Get driver info
+    	String driverName = type.getDriverName();
+    	serv.addEnvVar("DB_DRIVER", driverName);
+        
+        //Log results
+        Log.info(c, "setupDataSourceProperties", "Using database properties:"
+        		+ nl + "DriverName: " + driverName + nl + "DatabaseName: " + databaseName
+        		+ nl + "Username: " + username + nl + "Password: " + password 
+        		+ nl + "ServerName: " + serverName + nl + "Port: " + port);
+        
+        
+        //Get current config
+    	ServerConfiguration cloneConfig = serv.getServerConfiguration().clone();
+    	
+    	//Get Datasource config
+    	DataSource ds = cloneConfig.getDataSources().getById("DefaultDataSource");
+    	
+    	//Update properties based on type    	
+    	DataSourceProperties props = null;
+    	
+    	switch(type) {
+		case DB2:
+			props = new Properties_db2_jcc();
+			ds.clearDataSourceDBProperties();
+			ds.getProperties_db2_jcc().add((Properties_db2_jcc) props);
+			break;
+		case DERBY:
+			return; //Do not change
+		case INFORMIX:
+			props = new Properties_informix();
+			ds.clearDataSourceDBProperties();
+			ds.getProperties_informix().add((Properties_informix) props);
+			break;
+		case ORACLE:
+			Properties_oracle propsOracle = new Properties_oracle();
+			ds.clearDataSourceDBProperties();
+			ds.getProperties_oracle().add(propsOracle);
+			propsOracle.setUser(username);
+			propsOracle.setPassword(password);
+			propsOracle.setURL(cont.getJdbcUrl());
+			serv.updateServerConfiguration(cloneConfig);
+			return;
+		case POSTGRE:
+			props = new Properties();
+			ds.clearDataSourceDBProperties();
+			ds.getProperties().add((Properties) props);
+			break;
+		case SQLSERVER:
+			props = new Properties_microsoft_sqlserver();
+			ds.clearDataSourceDBProperties();
+			ds.getProperties_microsoft_sqlserver().add((Properties_microsoft_sqlserver) props);
+			break;
+		case SYBASE:
+			props = new Properties_sybase();
+			ds.clearDataSourceDBProperties();
+			ds.getProperties_sybase().add((Properties_sybase) props);
+			break;
+    	}
+    	
+    	props.setUser(username);
```

Comment:
I would prefer we don't use the server config API here, and instead let the FAT set environment variables on the server before it's started.

Reply:
I think that makes more sense for the case when we are testing against a single database test-container.  In that case you would want the flexibility to run tests using different data-source settings.  But, when doing database rotation I think this design works a better.  It eliminates the need for the person writing the test to worry about making sure the correct data-source properties are set for every database (since situations like DB2 arise where URL is not supported).  Instead they just worry about the test logic.

Reply:
I would also be interested to know if we have FAT test suites that use database rotation with multiple datasources.

Reply:
here are the hits I got (includes OL and CL) for usages of datasource modification. There are quite a few with multiple datasources, and also many that don't use `id="DefaultDataSource"`. At minimum, this method will need to accept one or more Strings for the dataSource ID's that should be updated, rather than assuming only the DefaultDataSource needs to be updated or that one exists.

```
./com.ibm.ws.jpa_spec10_injection_fat/publish/servers/JPAServerDFI/database.xml:    <dataSource id="jdbc/JPA_DS" jndiName="jdbc/JPA_DS" fat.modify="true">
./com.ibm.ws.jpa_spec10_injection_fat/publish/servers/JPAServerDFI/database.xml:    <dataSource id="jdbc/JPA_NJTADS" jndiName="jdbc/JPA_NJTADS" fat.modify="true" transactional="false">
./com.ibm.ws.jpa_spec10_injection_fat/publish/servers/JPAServer/database.xml:    <dataSource id="jdbc/JPA_DS" jndiName="jdbc/JPA_DS" fat.modify="true">
./com.ibm.ws.jpa_spec10_injection_fat/publish/servers/JPAServer/database.xml:    <dataSource id="jdbc/JPA_NJTADS" jndiName="jdbc/JPA_NJTADS" fat.modify="true" transactional="false">
./com.ibm.ws.jpa_spec10_injection_fat/publish/servers/JPAServerDMI/database.xml:    <dataSource id="jdbc/JPA_DS" jndiName="jdbc/JPA_DS" fat.modify="true">
./com.ibm.ws.jpa_spec10_injection_fat/publish/servers/JPAServerDMI/database.xml:    <dataSource id="jdbc/JPA_NJTADS" jndiName="jdbc/JPA_NJTADS" fat.modify="true" transactional="false">
./com.ibm.ws.security.javaeesec_fat/publish/servers/com.ibm.ws.security.javaeesec.db.anno.fat/server.xml:	<dataSource id="DefaultDataSource" fat.modify="true">
./com.ibm.ws.security.javaeesec_fat/publish/servers/com.ibm.ws.security.javaeesec.db.shortname.fat/server.xml:	<dataSource id="DefaultDataSource" fat.modify="true">
./com.ibm.ws.security.javaeesec_fat/publish/servers/com.ibm.ws.security.javaeesec.db.authdata.fat/server.xml:	<dataSource id="derby1" jndiName="jdbc/derby1fat" containerAuthDataRef="DBAuth"  fat.modify="true" >
./com.ibm.ws.security.javaeesec_fat/publish/servers/com.ibm.ws.security.javaeesec.dbidstore.deferred.fat/server.xml:	<dataSource id="DefaultDataSource" fat.modify="true">
./com.ibm.ws.security.javaeesec_fat/publish/servers/com.ibm.ws.security.javaeesec.dbidstore.deferred.fat/server.xml:    <dataSource id="NoDatabase" fat.modify="true" jndiName="jdbc/NoDatabase">
./com.ibm.ws.security.javaeesec_fat/publish/servers/com.ibm.ws.security.javaeesec.db.fat/server.xml:	<dataSource id="DefaultDataSource" fat.modify="true">
./com.ibm.ws.security.javaeesec_fat/publish/servers/com.ibm.ws.security.javaeesec.dbidstore.immediate.fat/server.xml:	<dataSource id="DefaultDataSource" fat.modify="true">
./com.ibm.ws.security.javaeesec_fat/publish/servers/com.ibm.ws.security.javaeesec.dbidstore.immediate.fat/server.xml:    <dataSource id="NoDatabase" fat.modify="true" jndiName="jdbc/NoDatabase">
./com.ibm.ws.security.javaeesec_fat/publish/servers/com.ibm.ws.security.javaeesec.db.authalias.fat/server.xml:	<dataSource id="derby1" jndiName="jdbc/derby1fat" fat.modify="true">
./com.ibm.ws.security.javaeesec_fat/publish/servers/com.ibm.ws.security.javaeesec.db.nouser.fat/server.xml:	<dataSource id="DefaultDataSource" fat.modify="true">
./com.ibm.ws.security.javaeesec_fat/publish/servers/com.ibm.ws.security.javaeesec.db.hash.bean.fat/server.xml:	<dataSource id="DefaultDataSource" fat.modify="true">
./com.ibm.ws.security.javaeesec_fat/publish/servers/com.ibm.ws.security.javaeesec.db.hash.fat/server.xml:	<dataSource id="DefaultDataSource" fat.modify="true">
./com.ibm.ws.jpa_spec10_fat/publish/servers/JPA10Server/database.xml:    <dataSource id="jdbc/JPA_DS" jndiName="jdbc/JPA_DS" fat.modify="true">
./com.ibm.ws.jpa_spec10_fat/publish/servers/JPA10Server/database.xml:    <dataSource id="jdbc/JPA_NJTADS" jndiName="jdbc/JPA_NJTADS" fat.modify="true" transactional="false">
./com.ibm.ws.jdbc_fat_v41/publish/servers/com.ibm.ws.jdbc.fat.v41/server.xml:    <dataSource id="DefaultDataSource" fat.modify="true">
./com.ibm.ws.jdbc_fat_v41/publish/servers/com.ibm.ws.jdbc.fat.v41/server.xml:    <dataSource id="ds1" jndiName="jdbc/${id}" fat.modify="true">
./com.ibm.ws.jdbc_fat_v41/publish/servers/com.ibm.ws.jdbc.fat.v41/server.xml:    <dataSource id="ds2" jndiName="jdbc/${id}" fat.modify="true">
./com.ibm.ws.jdbc_fat_v41/publish/servers/com.ibm.ws.jdbc.fat.v41/server.xml:    <dataSource id="ds3" jndiName="jdbc/${id}" fat.modify="true">
./com.ibm.ws.jdbc_fat_v41/publish/servers/com.ibm.ws.jdbc.fat.v41/server.xml:    <dataSource id="XAds" jndiName="jdbc/${id}" type="javax.sql.XADataSource" fat.modify="true">
./com.ibm.ws.jpa_spec20_fat/publish/servers/JPA20Server/database.xml:    <dataSource id="jdbc/JPA_DS" jndiName="jdbc/JPA_DS" fat.modify="true">
./com.ibm.ws.jpa_spec20_fat/publish/servers/JPA20Server/database.xml:    <dataSource id="jdbc/JPA_NJTADS" jndiName="jdbc/JPA_NJTADS" fat.modify="true" transactional="false">
./com.ibm.ws.jpa_22_fat/publish/servers/JPA22QueryServer/server.xml:    <dataSource id="DefaultDataSource" fat.modify="true">
./com.ibm.ws.jpa_22_fat/publish/servers/JPA22InjectionServer/server.xml:    <dataSource id="DefaultDataSource" fat.modify="true">
./com.ibm.ws.jpa_22_fat/publish/servers/JPA22InjectionServer/server.xml:    <dataSource id="jdbc/NonTransactionalDataSource" jndiName="jdbc/NonTransactionalDataSource" fat.modify="true" transactional="false">
./com.ibm.ws.jpa_22_fat/publish/servers/CDIFatServer/server.xml:    <dataSource id="DefaultDataSource" fat.modify="true">
./com.ibm.ws.jpa_22_fat/publish/servers/JPA22TimeAPIServer/server.xml:    <dataSource id="DefaultDataSource" fat.modify="true">
./com.ibm.ws.jpa_22_fat/publish/servers/JPABootstrapFATServer/server.xml:    <dataSource id="DefaultDataSource" fat.modify="true">
./com.ibm.ws.jpa_22_fat/publish/servers/EJBPassivationServer/server.xml:    <dataSource id="jdbc/JPA_DS" jndiName="jdbc/JPA_DS" fat.modify="true">
./com.ibm.ws.jpa_22_fat/publish/servers/EJBPassivationServer/server.xml:    <dataSource id="jdbc/JPA_NJTADS" jndiName="jdbc/JPA_NJTADS" fat.modify="true" transactional="false">
./com.ibm.ws.jpa_22_fat/publish/servers/JPA22BeanValidationServer/server.xml:    <dataSource id="DefaultDataSource" jndiName="jdbc/TransactionalDataSource" fat.modify="true">
./com.ibm.ws.jpa_22_fat/publish/servers/JPA22BeanValidationServer/server.xml:    <dataSource id="jdbc/NonTransactionalDataSource" jndiName="jdbc/NonTransactionalDataSource" fat.modify="true" transactional="false">
./com.ibm.ws.jpa.container.ormdiagnostics_fat/publish/servers/JPADiagTestServer/server.xml:    <dataSource id="DefaultDataSource" fat.modify="true">
./com.ibm.ws.jpa.container.ormdiagnostics_fat/build/libs/autoFVT/publish/servers/JPADiagTestServer/server.xml:    <dataSource id="DefaultDataSource" fat.modify="true">
./com.ibm.ws.jpa_spec10_query_fat/publish/servers/JPAQueryServer/database.xml:    <dataSource id="jdbc/JPA_DS" jndiName="jdbc/JPA_DS" fat.modify="true">
./com.ibm.ws.jpa_spec10_query_fat/publish/servers/JPAQueryServer/database.xml:    <dataSource id="jdbc/JPA_NJTADS" jndiName="jdbc/JPA_NJTADS" fat.modify="true" transactional="false">


./com.ibm.ws.jpa_fat/publish/servers/com.ibm.ws.jpa.fat.embeddable/server.xml:    <dataSource jndiName="jdbc/JPA_DS" jdbcDriverRef="DerbyEmbedded" fat.modify="true">
./com.ibm.ws.jpa_fat/publish/servers/com.ibm.ws.jpa.fat.embeddable/server.xml:    <dataSource jndiName="jdbc/JPA_NJTADS" jdbcDriverRef="DerbyEmbedded" transactional="false" fat.modify="true">
./com.ibm.ws.jpa_fat/publish/servers/com.ibm.ws.jpa.fat.emlocking/server.xml:	<dataSource jndiName="jdbc/JPA_DS" jdbcDriverRef="DerbyEmbedded" fat.modify="true" type="javax.sql.XADataSource">
./com.ibm.ws.jpa_fat/publish/servers/com.ibm.ws.jpa.fat.emlocking/server.xml:	<dataSource jndiName="jdbc/JPA_NJTADS" jdbcDriverRef="DerbyEmbedded" transactional="false" fat.modify="true">
./com.ibm.ws.jpa_fat/publish/servers/com.ibm.ws.jpa.fat.dsoverride/server.xml:    <dataSource jndiName="jdbc/JPA_DS" jdbcDriverRef="DerbyEmbedded" fat.modify="true" type="javax.sql.XADataSource">
./com.ibm.ws.jpa_fat/publish/servers/com.ibm.ws.jpa.fat.dsoverride/server.xml:    <dataSource jndiName="jdbc/AltJTADataSource" jdbcDriverRef="DerbyEmbedded" fat.modify="true" type="javax.sql.XADataSource">
./com.ibm.ws.jpa_fat/publish/servers/com.ibm.ws.jpa.fat.cache/server.xml:	<dataSource jndiName="jdbc/JPA_DS" jdbcDriverRef="DerbyEmbedded" fat.modify="true">
./com.ibm.ws.jpa_fat/publish/servers/com.ibm.ws.jpa.fat.cache/server.xml:	<dataSource jndiName="jdbc/JPA_NJTADS" jdbcDriverRef="DerbyEmbedded" transactional="false" fat.modify="true">
./com.ibm.ws.jpa_fat/publish/servers/com.ibm.ws.jpa.fat.ejbpassivation/server.xml:    <dataSource jndiName="jdbc/JPA_DS" jdbcDriverRef="DerbyEmbedded" fat.modify="true" type="javax.sql.XADataSource">
./com.ibm.ws.jpa_fat/publish/servers/com.ibm.ws.jpa.fat.entitymanager20/server.xml:	<dataSource jndiName="jdbc/JPA_DS" jdbcDriverRef="DerbyEmbedded" fat.modify="true" type="javax.sql.XADataSource">
./com.ibm.ws.jpa_fat/publish/servers/com.ibm.ws.jpa.fat.entitymanager20/server.xml:	<dataSource jndiName="jdbc/JPA_NJTADS" jdbcDriverRef="DerbyEmbedded" transactional="false" fat.modify="true">
./com.ibm.ws.jpa_fat/publish/servers/com.ibm.ws.jpa.fat.fetchgroup/server.xml:	<dataSource jndiName="jdbc/JPA_DS" jdbcDriverRef="DerbyEmbedded" fat.modify="true" type="javax.sql.XADataSource">
./com.ibm.ws.jpa_fat/publish/servers/com.ibm.ws.jpa.fat.fetchgroup/server.xml:	<dataSource jndiName="jdbc/JPA_NJTADS" jdbcDriverRef="DerbyEmbedded" transactional="false" fat.modify="true">
./com.ibm.ws.jpa_fat/publish/servers/com.ibm.ws.jpa.fat.ojexternalization/server.xml:	<dataSource jndiName="jdbc/JPA_DS" jdbcDriverRef="DerbyEmbedded" fat.modify="true" type="javax.sql.XADataSource">
./com.ibm.ws.jpa_fat/publish/servers/com.ibm.ws.jpa.fat.ojexternalization/server.xml:	<dataSource jndiName="jdbc/JPA_NJTADS" jdbcDriverRef="DerbyEmbedded" transactional="false" fat.modify="true">
./com.ibm.ws.jpa_fat/publish/servers/com.ibm.ws.jpa.fat.criteriaquery/server.xml:	<dataSource jndiName="jdbc/JPA_DS" jdbcDriverRef="DerbyEmbedded" fat.modify="true">
./com.ibm.ws.jpa_fat/publish/servers/com.ibm.ws.jpa.fat.criteriaquery/server.xml:	<dataSource jndiName="jdbc/JPA_NJTADS" jdbcDriverRef="DerbyEmbedded" transactional="false" fat.modify="true">
./com.ibm.ws.jdbc_fat/publish/servers/com.ibm.ws.jdbc.jaas.fat/server.xml:    <dataSource id="dsfatboot" jndiName="jdbc/dsfatboot" fat.modify="true" jdbcDriverRef="FATJDBCDriver" type="javax.sql.ConnectionPoolDataSource" transactional="false">
./com.ibm.ws.jdbc_fat/publish/servers/com.ibm.ws.jdbc.jaas.fat/server.xml:    <dataSource id="dsfatboot1" jndiName="jdbc/dsfatboot1" fat.modify="true" jdbcDriverRef="FATJDBCDriver1" type="javax.sql.ConnectionPoolDataSource" transactional="false">
./com.ibm.ws.jdbc_fat/publish/servers/com.ibm.ws.jdbc.jaas.fat/server.xml:    <dataSource id="dsfatboot2" jndiName="jdbc/dsfatboot2" fat.modify="true" jdbcDriverRef="FATJDBCDriver2" type="javax.sql.ConnectionPoolDataSource" transactional="false">
./com.ibm.ws.jdbc_fat/publish/servers/com.ibm.ws.jdbc.jaas.fat/server.xml:    <dataSource id="dsfatboot3" jndiName="jdbc/dsfatboot3" fat.modify="true" jdbcDriverRef="FATJDBCDriver3" type="javax.sql.ConnectionPoolDataSource" transactional="false">
./com.ibm.ws.jdbc_fat/publish/servers/com.ibm.ws.jdbc.jaas.fat/server.xml:    <dataSource id="dsfatboot4" jndiName="jdbc/dsfatboot4" fat.modify="true" type="javax.sql.ConnectionPoolDataSource" transactional="false">
./com.ibm.ws.jdbc_fat/publish/servers/com.ibm.ws.jdbc.jaas.fat/server.xml:    <dataSource id="dsfatboot5" jndiName="jdbc/dsfatboot5" fat.modify="true" type="javax.sql.ConnectionPoolDataSource" transactional="false">
./com.ibm.ws.jdbc_fat/publish/servers/com.ibm.ws.jdbc.jaas.fat/server.xml:    <dataSource id="dsfatboot6" jndiName="jdbc/dsfatboot6" fat.modify="true" type="javax.sql.ConnectionPoolDataSource" transactional="false">
./com.ibm.ws.jdbc_fat/publish/servers/com.ibm.ws.jdbc.jaas.fat/server.xml:    <dataSource id="dsfat0" jndiName="jdbc/dsfat0" fat.modify="true" jdbcDriverRef="FATJDBCDriver" type="javax.sql.DataSource" connectionSharing="MatchOriginalRequest">
./com.ibm.ws.jdbc_fat/publish/servers/com.ibm.ws.jdbc.jaas.fat/server.xml:    <dataSource id="dsfat1" jndiName="jdbc/dsfat1" fat.modify="true" jdbcDriverRef="FATJDBCDriver" connectionManagerRef="conMgr1" connectionSharing="MatchCurrentState" transactional="true">
./com.ibm.ws.jdbc_fat/publish/servers/com.ibm.ws.jdbc.jaas.fat/server.xml:    <dataSource id="dsfat2" jndiName="jdbc/dsfat2" fat.modify="true" jdbcDriverRef="FATJDBCDriver" connectionManagerRef="conMgr2" type="javax.sql.XADataSource" isolationLevel="TRANSACTION_READ_COMMITTED" recoveryAuthDataRef="recoveryAuth">
./com.ibm.ws.jdbc_fat/publish/servers/com.ibm.ws.jdbc.jaas.fat/server.xml:    <dataSource id="dsfat3" jndiName="jdbc/dsfat3" fat.modify="true" jdbcDriverRef="FATJDBCDriver" type="javax.sql.ConnectionPoolDataSource" transactional="false">
./com.ibm.ws.jdbc_fat/publish/servers/com.ibm.ws.jdbc.jaas.fat/server.xml:    <dataSource jndiName="jdbc/dsfat4" fat.modify="true" jdbcDriverRef="FATJDBCDriver" type="javax.sql.XADataSource" transactional="true" connectionSharing="MatchCurrentState" recoveryAuthDataRef="recoveryAuth">
./com.ibm.ws.jdbc_fat/publish/servers/com.ibm.ws.jdbc.jaas.fat/server.xml:    <dataSource id="dsfat20" fat.modify="true" jndiName="jdbc/dsfat20" jdbcDriverRef="FATJDBCDriver" connectionManagerRef="conMgr6" type="javax.sql.XADataSource" isolationLevel="TRANSACTION_READ_COMMITTED" recoveryAuthDataRef="recoveryAuth">
./com.ibm.ws.jdbc_fat/publish/servers/com.ibm.ws.jdbc.jaas.fat/server.xml:    <dataSource id="dsfat21" fat.modify="true" jndiName="jdbc/dsfat21" jdbcDriverRef="FATJDBCDriver" type="javax.sql.ConnectionPoolDataSource">
./com.ibm.ws.jdbc_fat/publish/servers/com.ibm.ws.jdbc.jaas.fat/server.xml:    <dataSource id="dsfat22" fat.modify="true" jndiName="jdbc/dsfat22" jdbcDriverRef="FATJDBCDriver" type="javax.sql.ConnectionPoolDataSource">
./com.ibm.ws.jdbc_fat/publish/servers/com.ibm.ws.jdbc.jaas.fat/server.xml:    <dataSource id="dsfat23" fat.modify="true" jndiName="jdbc/dsfat23" jdbcDriverRef="FATJDBCDriver" type="javax.sql.ConnectionPoolDataSource">
./com.ibm.ws.jdbc_fat/publish/servers/com.ibm.ws.jdbc.fat/server.xml:    <dataSource id="dsfatboot" jndiName="jdbc/dsfatboot" fat.modify="true" jdbcDriverRef="FATJDBCDriver" type="javax.sql.ConnectionPoolDataSource" transactional="false">
./com.ibm.ws.jdbc_fat/publish/servers/com.ibm.ws.jdbc.fat/server.xml:    <dataSource id="dsfatboot1" jndiName="jdbc/dsfatboot1" fat.modify="true" jdbcDriverRef="FATJDBCDriver1" type="javax.sql.ConnectionPoolDataSource" transactional="false">
./com.ibm.ws.jdbc_fat/publish/servers/com.ibm.ws.jdbc.fat/server.xml:    <dataSource id="dsfatboot2" jndiName="jdbc/dsfatboot2" fat.modify="true" jdbcDriverRef="FATJDBCDriver2" type="javax.sql.ConnectionPoolDataSource" transactional="false">
./com.ibm.ws.jdbc_fat/publish/servers/com.ibm.ws.jdbc.fat/server.xml:    <dataSource id="dsfatboot3" jndiName="jdbc/dsfatboot3" fat.modify="true" jdbcDriverRef="FATJDBCDriver3" type="javax.sql.ConnectionPoolDataSource" transactional="false">
./com.ibm.ws.jdbc_fat/publish/servers/com.ibm.ws.jdbc.fat/server.xml:    <dataSource id="dsfatboot4" jndiName="jdbc/dsfatboot4" fat.modify="true" type="javax.sql.ConnectionPoolDataSource" transactional="false">
./com.ibm.ws.jdbc_fat/publish/servers/com.ibm.ws.jdbc.fat/server.xml:    <dataSource id="dsfatboot5" jndiName="jdbc/dsfatboot5" fat.modify="true" type="javax.sql.ConnectionPoolDataSource" transactional="false">
./com.ibm.ws.jdbc_fat/publish/servers/com.ibm.ws.jdbc.fat/server.xml:    <dataSource id="dsfatboot6" jndiName="jdbc/dsfatboot6" fat.modify="true" type="javax.sql.ConnectionPoolDataSource" transactional="false">
./com.ibm.ws.jdbc_fat/publish/servers/com.ibm.ws.jdbc.fat/server.xml:    <dataSource id="dsfat0" jndiName="jdbc/dsfat0" fat.modify="true" jdbcDriverRef="FATJDBCDriver" type="javax.sql.DataSource" connectionSharing="MatchOriginalRequest">
./com.ibm.ws.jdbc_fat/publish/servers/com.ibm.ws.jdbc.fat/server.xml:    <dataSource id="dsfat1" jndiName="jdbc/dsfat1" fat.modify="true" jdbcDriverRef="FATJDBCDriver" connectionManagerRef="conMgr1" connectionSharing="MatchCurrentState" transactional="true">
./com.ibm.ws.jdbc_fat/publish/servers/com.ibm.ws.jdbc.fat/server.xml:    <dataSource id="dsfat2" jndiName="jdbc/dsfat2" fat.modify="true" jdbcDriverRef="FATJDBCDriver" connectionManagerRef="conMgr2" type="javax.sql.XADataSource" isolationLevel="TRANSACTION_READ_COMMITTED" recoveryAuthDataRef="recoveryAuth">
./com.ibm.ws.jdbc_fat/publish/servers/com.ibm.ws.jdbc.fat/server.xml:    <dataSource id="dsfat3" jndiName="jdbc/dsfat3" fat.modify="true" jdbcDriverRef="FATJDBCDriver" type="javax.sql.ConnectionPoolDataSource" transactional="false">
./com.ibm.ws.jdbc_fat/publish/servers/com.ibm.ws.jdbc.fat/server.xml:    <dataSource jndiName="jdbc/dsfat4" fat.modify="true" jdbcDriverRef="FATJDBCDriver" type="javax.sql.XADataSource" transactional="true" connectionSharing="MatchCurrentState" recoveryAuthDataRef="recoveryAuth">
./com.ibm.ws.jdbc_fat/publish/servers/com.ibm.ws.jdbc.fat/server.xml:    <dataSource id="dsfat20" fat.modify="true" jndiName="jdbc/dsfat20" jdbcDriverRef="FATJDBCDriver" connectionManagerRef="conMgr6" type="javax.sql.XADataSource" isolationLevel="TRANSACTION_READ_COMMITTED" recoveryAuthDataRef="recoveryAuth">
./com.ibm.ws.jdbc_fat/publish/servers/com.ibm.ws.jdbc.fat/server.xml:    <dataSource id="dsfat21" fat.modify="true" jndiName="jdbc/dsfat21" jdbcDriverRef="FATJDBCDriver" type="javax.sql.ConnectionPoolDataSource">
./com.ibm.ws.jdbc_fat/publish/servers/com.ibm.ws.jdbc.fat/server.xml:    <dataSource id="dsfat22" fat.modify="true" jndiName="jdbc/dsfat22" jdbcDriverRef="FATJDBCDriver" type="javax.sql.ConnectionPoolDataSource">
./com.ibm.ws.jdbc_fat/publish/servers/com.ibm.ws.jdbc.fat/server.xml:    <dataSource id="dsfat23" fat.modify="true" jndiName="jdbc/dsfat23" jdbcDriverRef="FATJDBCDriver" type="javax.sql.ConnectionPoolDataSource">
./com.ibm.ws.jdbc_fat_v42/publish/servers/jdbc.fat.v42/server.xml:    <dataSource id="DefaultDataSource" fat.modify="true">
./com.ibm.ws.jpa_fat_java8/publish/servers/com.ibm.ws.jpa.fat.java8/server.xml:	<dataSource jndiName="jdbc/JPA_DS" jdbcDriverRef="DerbyEmbedded" fat.modify="true">
./com.ibm.ws.jpa_fat_java8/publish/servers/com.ibm.ws.jpa.fat.java8/server.xml:	<dataSource jndiName="jdbc/JPA_NJTADS" jdbcDriverRef="DerbyEmbedded" transactional="false" fat.modify="true">
./com.ibm.ws.concurrent.persistent_fat_timers/publish/servers/com.ibm.ws.concurrent.persistent.fat.timers/server.xml:  <dataSource id="DefaultDataSource" fat.modify="true">
./com.ibm.ws.jdbc_fat_fvt/publish/servers/com.ibm.ws.jdbc.fat.fvt/server.xml:    <dataSource id="basic-datasource" jndiName="jdbc/${id}" type="javax.sql.ConnectionPoolDataSource" fat.modify="true">
./com.ibm.ws.jdbc_fat_fvt/publish/servers/com.ibm.ws.jdbc.fat.fvt/server.xml:    <dataSource id="basic-datasource2" jndiName="jdbc/${id}" type="javax.sql.ConnectionPoolDataSource" fat.modify="true">
./com.ibm.ws.concurrent.persistent_fat_multiple/publish/servers/com.ibm.ws.concurrent.persistent.fat.multiple/server.xml:  <dataSource id="DefaultDataSource" fat.modify="true" containerAuthDataRef="authx">
```